exercises/square-root/instructions.md

```diff
@@ -0,0 +1,18 @@
+# Instructions
+
+Your task is to calculate the square root of a given number.
+
+- Try to avoid using the pre-existing math libraries of your language.
+- As input you’ll be given a positive whole number, i.e. 1, 2, 3, 4…
+- You also only need to output a positive whole number.
+
+Here are some approaches you could use.
+
+- Linear or binary search for a number that gives the input number when squared.
+- Successive approximation using Newton's or Heron's method.
+- Calculating one digit at a time or one bit at a time.
+
+You can check out the Wikipedia pages on [integer square root][integer-square-root] and [methods of computing square roots][computing-square-roots] to help with choosing a method of calculation.
+
+[integer-square-root]: https://en.wikipedia.org/wiki/Integer_square_root
```

Comment:
```suggestion
```

The integer square root differs from the usual square root, and the existing wiki link already provides enough information. It might lead students down an unintended path and cause extra confusion.

Reply:
I thought the idea was to have students specifically implement integer square root, not the full floating point square root, as a way of narrowing down the scope of the task.  The backstory was chosen partly to support this.

Reply:
Additional reasons why I intentionally want to explicitly require the integer square root in the problem specification:

* Algorithms that compute the non-integer square root run forever on inputs that are not a perfect square.  The integer square root algorithms do not run forever.
* The existing submissions, at least in the Python track, have taken their cues from the tests expecting integer results and the students have typically truncated the results to an integer even when using the built in operators, for example many submissions are of the form int(n^0.5).
* The integer square root has a wider range of alternative implementations as suggested in these new instructions, which I think makes the exercise more interesting.

Therefore if any Wikipedia link is to be removed I would prefer it be the one to the "Computing square roots" page, not the "Integer square roots" page.  The inverse of your requested change.

Reply:
I'm with @xanni on this one. Judging from the original implementation, the author meant this exercise as an exercise on perfect squares.
I also thin that both links can stay. Not that many people will bother following up two links before solving the exercise. If they get stuck, they loo to community solutions instead.

Reply:
Not to mention the risk of them doing their own research and being exposed to things.

Reply:
Wikipedia links removed.